gammapy/datasets/map.py

```diff
@@ -337,20 +337,35 @@ def data_shape(self):
         """Shape of the counts or background data (tuple)"""
         return self._geom.data_shape
 
-    def _energy_range(self, mask = None):
-        """Compute the energy range with or without the mask fit."""
+    # TODO: make this support different methods?
```

Comment:
I do not understand the meaning of this ToDo? Which kind of different methods?

Reply:
I think the idea here was to maybe support different methods like `.energy_range(which="safe")`. This is bit more flexible than introducing multiple attributes like `.energy_range_safe` and `.energy_range_fit`...

Reply:
Since I had to adapt this comment a long time ago, it remains as a change and is carried in diffs to this date even though it was already removed from master.